action.yml

```
@@ -1,12 +1,212 @@
+        RESP_CODE=$(curl -w %{http_code} -s -L -o __tags.json \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer ${{ github.token }}" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          https://api.github.com/repos/silverstripe/silverstripe-framework/tags
+        )
+        if [[ $RESP_CODE != "200" ]]; then
+          echo "Failed to get list of tags - HTTP response code was $RESP_CODE"
+          cat __tags.json
+          exit 1
+        fi
+        # Check if there is y.2.0 tag for the current supported major version of framework
+        FRAMEWORK_TWO_TAG_EXISTS=0
+        TWO_TAG=$(jq -r --arg framework_supported_major "$FRAMEWORK_SUPPORTED_MAJOR" '.[] | select(.name == $framework_supported_major + ".2.0")' __tags.json)
+        if ! [[ -z $TWO_TAG ]]; then
+          FRAMEWORK_TWO_TAG_EXISTS=1
+        fi
+        echo "FRAMEWORK_TWO_TAG_EXISTS is $FRAMEWORK_TWO_TAG_EXISTS"
+        echo "framework_two_tag_exists=$FRAMEWORK_TWO_TAG_EXISTS" >> "$GITHUB_OUTPUT"
+        # Check if there is y.4.0 tag for the current supported major version of framework
+        FRAMEWORK_FOUR_TAG_EXISTS=0
+        FOUR_TAG=$(jq -r --arg framework_supported_major "$FRAMEWORK_SUPPORTED_MAJOR" '.[] | select(.name == $framework_supported_major + ".4.0")' __tags.json)
+        if ! [[ -z $FOUR_TAG ]]; then
+          FRAMEWORK_FOUR_TAG_EXISTS=1
+        fi
+        echo "FRAMEWORK_FOUR_TAG_EXISTS is $FRAMEWORK_FOUR_TAG_EXISTS"
+        echo "framework_four_tag_exists=$FRAMEWORK_FOUR_TAG_EXISTS" >> "$GITHUB_OUTPUT"
+        rm __tags.json
+
+    - name: Derive module branch
+      id: derive-module-branch
+      shell: bash
+      env:
+        GITHUB_REPOSITORY: ${{ github.repository }}
+        DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+        BRANCH_TYPE: ${{ steps.derive-branch-type.outputs.branch_type }}
+        FRAMEWORK_SUPPORTED_MAJOR: ${{ steps.derive-supported-framwork.outputs.framework_supported_major }}
+        FRAMEWORK_TWO_TAG_EXISTS: ${{ steps.derive-supported-framwork.outputs.framework_two_tag_exists }}
+        FRAMEWORK_FOUR_TAG_EXISTS: ${{ steps.derive-supported-framwork.outputs.framework_four_tag_exists }}
+      run: |
+        # $DEFAULT_BRANCH is always assumed to be a numeric value i.e. not "main", and is assumed to be the same CMS version as $FRAMEWORK_SUPPORTED_MAJOR
+        # It's always assumed that this action is run every six months, and at least one month from a minor beta release
+        # meaning that whenever we get the highest minor branch for the current minor, that minor branch will currently be stable
+        echo "DEFAULT_BRANCH is $DEFAULT_BRANCH"
+        MAJOR=""
+        if [[ "$BRANCH_TYPE" == "prev-major-curr-minor" ]]; then
+          if [[ "$FRAMEWORK_TWO_TAG_EXISTS" == "1" ]]; then
+            echo "Because framework tag $FRAMEWORK_SUPPORTED_MAJOR.2.0 exists, the 'prev-major-curr-minor' run type is no longer supported"
```

```suggestion
            echo "The previous major release line no longer has bug fix support, so the 'prev-major-curr-minor' run type is no longer supported"
```
This means we don't have to mentally translate "x.2.0 exists" into "no bug fix support" because it's just spelled out directly.

---

This does assume that there are no security problems in any of the JS though... or are we relying on dependabot for those? Or just ignoring them unless someone reports them?

This probably becomes a Jenn question.

---

> This probably becomes a Jenn question.

Have discussed offline with Jenn. Will disable cron at 6.2.0 framework tag. Any JS security vulns for partially supported versions won't be managed using this particular github action.

---

Have updated echo statement